extensions/mssql/src/ui/dialogBase.ts

```diff
@@ -147,6 +147,25 @@ export abstract class DialogBase<DialogResult> {
 		return this.createInputBox(ariaLabel, textChangeHandler, value, enabled, 'password', width);
 	}
 
+	protected createInputBoxWithProperties(textChangeHandler: (newValue: string) => Promise<void>, properties: azdata.InputBoxProperties, customValidation?: () => Promise<boolean>): azdata.InputBoxComponent {
+		properties.width = properties.width ?? DefaultInputWidth;
+		properties.inputType = properties.inputType ?? 'text';
+		properties.value = properties.value ?? '';
+		properties.enabled = properties.enabled ?? true;
+		let inputbox: azdata.InputBoxComponent;
+		if (customValidation) {
+			inputbox = this.modelView.modelBuilder.inputBox().withProps(properties).withValidation(customValidation).component();
```

Comment:
To avoid duplication and chance of errors you can do something like this.

```
const inputBox = this.modelView.modelBuilder.inputBox().withProps(properties);
if (customValidation) {
   inputBox.withValidation(customValidation);
}
const inputBoxComponent = inputBox.component();